### PR TITLE
Books/fix relative links

### PR DESCRIPTION
**Pull request by MaddyDev**
Fix for #6973 

Relative links don't work on untitled files. 
Fix: Opening them with untitled scheme works on Mac but for windows the entire path of the resource file with untitled scheme is needed. (TO DO: Need to test on Linux - hoping the mac way would work it too.)

### CODE REVIEW

File: src/sql/workbench/parts/notebook/electron-browser/outputs/notebookMarkdown.ts [hunk 1]
```diff
@@ -226,6 +226,8 @@ export class NotebookMarkdownRenderer {
 			return base.replace(/:[\s\S]*/, ':') + href;
 		} else if (href.charAt(0) === '/') {
 			return base.replace(/(:\/*[^/]*)[\s\S]*/, '$1') + href;
+		} else if (href.slice(0, 2) === '..') {
+			return path.join(base, href);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/notebook/electron-browser/outputs/notebookMarkdown.ts]: What about a single . ?

File: extensions/notebook/src/book/bookTreeView.ts [hunk 2]
```diff
@@ -332,19 +332,25 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
...
 		return this._tableOfContentPaths;
 	}
+
 	getUntitledNotebookUri(resource: string): vscode.Uri {
-		let title = this.findNextUntitledFileName(resource);
-		let untitledFileName: vscode.Uri = vscode.Uri.parse(`untitled:${title}`);
-		if (!this._allNotebooks.get(untitledFileName.fsPath)) {
+		let untitledFileName: vscode.Uri;
+		if (process.platform.indexOf('win') > -1) {
+			let title = path.join(path.dirname(resource), this.findNextUntitledFileName(resource));
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts]: Why do it like this here? We've always just done process.platform === 'win32' - that's the only value returned for windows machines.

File: extensions/notebook/src/book/bookTreeView.ts [hunk 3]
```diff
@@ -332,19 +332,25 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
...
+			untitledFileName = vscode.Uri.parse(`untitled:${title}`);
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts]: >untitled: [](start = 40, length = 9)

Why not use with({ scheme: 'untitled' })?

Comment (MaddyDev) [reply extensions/notebook/src/book/bookTreeView.ts]: wish that works! I had that before. It only works on MAC that way but on windows it doesn't. the notebookUri is only preserved when I have it this way.

File: extensions/notebook/src/book/bookTreeView.ts (hunk 2 above)
Comment (MaddyDev) [reply extensions/notebook/src/book/bookTreeView.ts]: ohh, I didn't know that. I'll make that change. Thanks!

**Pull request merged**